bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/handler/MagentaTVHandler.java

```diff
@@ -0,0 +1,688 @@
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.NextPreviousType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PlayPauseType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.RewindFastforwardType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.magentatv.internal.MagentaTVDeviceManager;
+import org.openhab.binding.magentatv.internal.MagentaTVException;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.MRPayEvent;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.MRPayEventInstanceCreator;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.MRProgramInfoEvent;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.MRProgramInfoEventInstanceCreator;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.MRProgramStatus;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.MRProgramStatusInstanceCreator;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.MRShortProgramInfo;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.MRShortProgramInfoInstanceCreator;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.OAuthAuthenticateResponse;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.OAuthTokenResponse;
+import org.openhab.binding.magentatv.internal.MagentaTVGsonDTO.OauthCredentials;
+import org.openhab.binding.magentatv.internal.config.MagentaTVDynamicConfig;
+import org.openhab.binding.magentatv.internal.config.MagentaTVThingConfiguration;
+import org.openhab.binding.magentatv.internal.network.MagentaTVNetwork;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link MagentaTVHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+public class MagentaTVHandler extends BaseThingHandler implements MagentaTVListener {
+    private final Logger logger = LoggerFactory.getLogger(MagentaTVHandler.class);
+    private static final DecimalType ZERO = new DecimalType(0);
```

Comment:
zero is already predefined in `DecimalType`

Reply:
changed